broke-broker/src/main/java/org/dsngroup/broke/broker/storage/PingRequests.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2017 original authors and authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dsngroup.broke.broker.storage;
+
+/**
+ * Storage for PINGREQ messages
+ * */
+public class PingRequests {
+
+    private int pingReqSize = 100;
+
+    private PingReq[] pingReq;
+
+    public long getPingReq(int packetId) {
+        return pingReq[packetId % pingReqSize].getPingReq(packetId);
+    }
+
+    public void setPingReq(int packetId, long sendTime) {
+        pingReq[packetId % pingReqSize].setPingReq(packetId, sendTime);
+    }
+
+    public PingRequests() {
+        this(100);
+    }
+
+    public PingRequests(int pingReqSize) {
+        pingReq = new PingReq[pingReqSize];
+        for (int i=0; i<pingReqSize; i++) {
+            pingReq[i] = new PingReq();
+        }
+    }
+}
+
```

Comment:
IMHO, just store the **average** rtt and additional statical information for each clients(session), and figure out the way to truncate the non-normal time. 
i.e. assume a client is average at 20ms, and if the next rtt as 30ms is counted, will we store this? Just like the concept of standard deviation.

Reply:
Therefore, you should think of **HOW TO MAKE DECISION**, as a core algorithm.

Reply:
But how about the issue of client's condition change (e.g. mobile clients)? It is hard to tell that an abnormal rtt is caused by the condition change of client or just a packet loss, etc. 
And averaging the rtts may cause the condition change of clients hard or takes longer time to be awared by the broker server.
Of course, tradeoff can be considers. For example, only average a small amount of rtts.

Reply:
In the abnormal condition, as I mentioned, you can have a variable range for abandoned counting rtt, and this range depends on history and desired timeout.
Whatever policy you choose, the final result is to based on a **value (score)** to dispatch workload.